src/compiler/types.ts

```diff
@@ -2269,11 +2269,20 @@ namespace ts {
     
     export interface ResolvedModule {
         resolvedFileName: string;
+        /*
+         * Denotes if 'resolvedFileName' should be proper external module:
+         * - be a .d.ts file 
+         * - use top level imports\exports
+         * - don't use tripleslash references
+         */
+        shouldBeProperExternalModule?: boolean;

```

Comment:
i think this should be declarative. i.e. the name tells you what it is rather than telling you how it should be have. so something like
isExternalLibrary, or isExternalLibraryImport